ApeWorX/ape
# feat: support `request_header` driven from configs

### What I did

* Allow you to configure request headers across the board, via ecosystems, via networks, and via providers.
* Actually use said request_header in web3.py (was missing!)

### How to verify it

```yaml
request_headers:
  Top-Level: "UseThisOnEveryRequest"

ethereum:
  request_headers:
    Ecosystem-Level: "UseThisOnEveryEthereumRequest"
  
  mainnet:
    request_headers:
      Network-Level: "UseThisOnAllRequestsToEthereumMainnet"

node:
  request_headers:
    Provider-Level: "UseThisOnAllRequestsUsingNodeProvider"
```

then connect:

```sh
ape console --network ethereum:local:node
```

Then verify headers:

```python
In [12]: http = provider.web3.provider._potential_providers[2]()

In [13]: http._request_kwargs
Out[13]: 
{'headers': {'User-Agent': 'Ape/0.8.14.dev0+g228ab19a1.d20240827 (Python/3.12.2 final)',
  'Top-Level': 'UseThisOnEveryRequest',
  'Ecosystem-Level': 'UseThisOnEveryEthereumRequest',
  'Provider-Level': 'UseThisOnAllRequestsUsingNodeProvider'},
 'timeout': 1800}
```

### Checklist
